dsgrid/config/dimensions.py

```diff
@@ -271,15 +271,14 @@ def check_file(cls, filename):
     @validator("file_hash")
     def compute_file_hash(cls, file_hash, values):
         if "filename" not in values:
-            # TODO #194
-            # We are getting here for Time. That shouldn't be happening.
-            # This seems to work, but something is broken.
             return None
         return file_hash or compute_file_hash(values["filename"])
 
     @validator("records", always=True)
     def add_records(cls, records, values):
         """Add records from the file."""
+        if not {"filename", "cls", "name"}.issubset(values):
+            return None
```

Comment:
Just checking--is it okay/expected that calling this method might result in nothing happening? Seems like yes because this is a validator?

Reply:
We've had a lot of problems like this. Pydantic always runs all validators, which can be problematic if a validator uses the `values` dict. That dict gets populated with values from successfully validated fields. If a validator fails, that field will not be in the dict and so every future validator needs to handle the case. I wish that they had a fail-fast option.